Explorer/Assets/DCL/Backpack/BackpackGridController.cs

```diff
@@ -0,0 +1,157 @@
+                backpackItemView.FlapBackground.color = rarityColors.GetColor(gridWearables[i].GetRarity());
+                backpackItemView.CategoryImage.sprite = categoryIcons.GetTypeImage(gridWearables[i].GetCategory());
+
+                WaitForThumbnailAsync(gridWearables[i], backpackItemView).Forget();
+            }
+        }
+
+        private BackpackItemView CreateBackpackItem(BackpackItemView backpackItem)
+        {
+            BackpackItemView backpackItemView = Object.Instantiate(backpackItem, view.gameObject.transform);
+            backpackItem.EquipButton.onClick.AddListener(() => commandBus.SendCommand(new BackpackEquipCommand(backpackItemView.ItemId)));
+            return backpackItemView;
+        }
+
+        public void RequestPage(int pageNumber)
+        {
+            //Reuse params array and review types URLParameter once auth pr is merged
+            ParamPromise wearablesPromise = ParamPromise.Create(world, new GetWearableByParamIntention(new[] { ("pageNumber", string.Format("{0}", pageNumber)), ("pageSize", "16") },  web3IdentityCache.Identity!.EphemeralAccount.Address, new List<IWearable>()), PartitionComponent.TOP_PRIORITY);
+            AwaitWearablesPromiseAsync(wearablesPromise).Forget();
+        }
+
+        private async UniTaskVoid AwaitWearablesPromiseAsync(ParamPromise wearablesPromise)
+        {
+            AssetPromise<IWearable[],GetWearableByParamIntention> uniTaskAsync = await wearablesPromise.ToUniTaskAsync(world);
+
+            if (!uniTaskAsync.Result.Value.Succeeded)
+                return;
+
+            SetGridElements(uniTaskAsync.Result.Value.Asset);
+        }
+
+        private async UniTaskVoid WaitForThumbnailAsync(IWearable itemWearable, BackpackItemView itemView)
+        {
+            do
+            {
+                await UniTask.Delay(500);
+            }
+            while (itemWearable.WearableThumbnail == null);
+
+            Texture2D valueAsset = itemWearable.WearableThumbnail.Value.Asset;
+            itemView.WearableThumbnail.sprite = Sprite.Create(valueAsset, new Rect(0, 0, valueAsset.width, valueAsset.height), VectorUtilities.OneHalf, 50, 0, SpriteMeshType.FullRect, Vector4.one, false);
+        }
+
+        private void ClearPoolElements()
+        {
+            foreach (var backpackItemView in usedPoolItems)
+            {
+                backpackItemView.Value.OnSelectItem -= SelectItem;
+                backpackItemView.Value.EquippedIcon.SetActive(false);
+                gridItemsPool.Release(backpackItemView.Value);
+            }
+
+            usedPoolItems.Clear();
+        }
+
+        private void SelectItem(string itemId)
+        {
+            commandBus.SendCommand(new BackpackSelectCommand(itemId));
+        }
+
+        private void OnUnequip(IWearable unequippedWearable)
+        {
+            if (usedPoolItems.ContainsKey(unequippedWearable.GetUrn()))
```

Comment:
You can use `TryGetValue` instead of accessing the dictionary and calling `GetUrn` twice